sklearn/cluster/_hdbscan/hdbscan.py

```diff
@@ -116,30 +116,38 @@ def _hdbscan_brute(
     **metric_params,
 ):
     if metric == "precomputed":
-        # Treating this case explicitly, instead of letting
-        # sklearn.metrics.pairwise_distances handle it,
-        # enables the usage of numpy.inf in the distance
-        # matrix to indicate missing distance information.
+        if X.shape[0] != X.shape[1]:
+            raise ValueError(
```

Comment:
According to codecov, there are no tests for new `ValueError`s. Can you add them?